devtools: Support for Sets in component properties.

JeanMeche opened the pull request:
With this commit it is now possible to expand the Sets (like Arrays or Object) on the property list. 

Fixes #49312

AleksanderBodurri commented:
Hey @JeanMeche thanks for the PR. This looks good, the only thing that I think would keep this from landing is that currently there isn't a way to edit any of the values in a Set. I wonder if it would be better to simply display the size of the Set like you're currently doing in this PR, but without it being expandable. 

Since we already expose the currently selected directive/component through `$ng0` in the chrome devtools console, a user could fall back on that to inspect the values in a set if necessary. Let me know what you think 🙏

JeanMeche committed:
feat(devtools): Improve Set support in component properties.

With this commit, Sets are displayed with the label Set(#) where # is the size of the set. They are not expandable/editable though.

Partial fix for #49312

AleksanderBodurri reviewed (approved):
Thanks for this @JeanMeche 🙏 Looks great

dylhunn commented:
This PR was merged into the repository by commit ba3e9eac92231bfae6c1ea9b289fe5544a3bf7e0.
